src/OrbitQt/TargetLabelTest.cpp

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+
+#include <memory>
+
+#include "OrbitGgp/Instance.h"
+#include "TargetLabel.h"
+#include "process.pb.h"
+
+namespace orbit_qt {
+
+namespace fs = std::filesystem;
+
+TEST(TargetLabel, Constructor) {
+  TargetLabel label{};
+  EXPECT_TRUE(label.text().isEmpty());
+  EXPECT_TRUE(label.styleSheet().isEmpty());
+  EXPECT_TRUE(label.toolTip().isEmpty());
+}
+
+TEST(TargetLabel, SetFileTarget) {
+  TargetLabel label{};
+
+  const fs::path filename = "file.orbit";
+  const fs::path path = "dummy/path/" / filename;
+
+  label.SetFileTarget(path);
+
+  EXPECT_EQ(label.text().toStdString(), filename);
+  EXPECT_TRUE(label.styleSheet().contains(TargetLabel::kDefaultColor));
+  EXPECT_TRUE(label.toolTip().isEmpty());
+}
+
+TEST(TargetLabel, SetStadiaTarget) {
+  TargetLabel label{};
+
+  const QString process_name = "test process name";
+  const QString instance_name = "test instance name";
+
+  ProcessData process;
+  orbit_grpc_protos::ProcessInfo process_info{};
+  process_info.set_name(process_name.toStdString());
+  process.SetProcessInfo(process_info);
+  orbit_ggp::Instance instance{};
+  instance.display_name = instance_name;
+
+  label.SetStadiaTarget(process, instance);
+
+  EXPECT_EQ(label.text(), process_name + " @ " + instance_name);
+  EXPECT_TRUE(label.styleSheet().contains(TargetLabel::kGreenColor));
+  EXPECT_TRUE(label.toolTip().isEmpty());
+}
+
+TEST(TargetLabel, SetLocalTarget) {
+  TargetLabel label{};
+
+  const QString process_name = "test process name";
+
+  ProcessData process;
+  orbit_grpc_protos::ProcessInfo process_info{};
+  process_info.set_name(process_name.toStdString());
+  process.SetProcessInfo(process_info);
+
+  label.SetLocalTarget(process);
+  EXPECT_EQ(label.text(), process_name + " @ localhost");
+  EXPECT_TRUE(label.styleSheet().contains(TargetLabel::kGreenColor));
+  EXPECT_TRUE(label.toolTip().isEmpty());
+}
+
+TEST(TargetLabel, SetProcessDead) {
+  TargetLabel label;
+  label.SetProcessDead();
+
+  EXPECT_TRUE(label.styleSheet().contains(TargetLabel::kOrangeColor));
+  EXPECT_EQ(label.toolTip(), "The process ended on the instance");
+}
+
+TEST(TargetLabel, SetProcessAlive) {
+  TargetLabel label;
+  label.SetProcessAlive();
+
+  EXPECT_TRUE(label.styleSheet().contains(TargetLabel::kGreenColor));
+  EXPECT_TRUE(label.toolTip().isEmpty());
+}
+
+TEST(TargetLabel, SetConnectionDead) {
+  TargetLabel label;
+  const QString error_message{"dummy error message"};
+  label.SetConnectionDead(error_message);
+
+  EXPECT_TRUE(label.styleSheet().contains(TargetLabel::kRedColor));
+  EXPECT_EQ(label.toolTip(), error_message);
+}
+
+TEST(TargetLabel, StateChanging) {
+  TargetLabel label;
+
+  label.SetProcessAlive();
+  EXPECT_TRUE(label.styleSheet().contains(TargetLabel::kGreenColor));
+  EXPECT_TRUE(label.toolTip().isEmpty());
```

Comment:
Nit: Since you test this twice (here and in `SetProcessAlive`), it would be easier to maintain and update if you'd move it to a function. Same for the three tests below.

Reply:
outdated